lib/Dialect/SPIRV/SPIRVOps.cpp

```diff
@@ -1583,6 +1583,51 @@ static LogicalResult verify(spirv::LoadOp loadOp) {
   return verifyMemoryAccessAttribute(loadOp);
 }
 
+//===----------------------------------------------------------------------===//
```

Comment:
Actually we can use tblgen to define these patterns concisely. But given you've already implemented this, okay for me to land this first and the improve later. :)

Reply:
@antiagainst thanks for mention this, definitely I have to learn tblgen style for "match/rewrite patterns".

Reply:
FYI: I've landed https://github.com/tensorflow/mlir/commit/c3ae3aee69e97211277e50ef68c5a5ac192da9ca that simplifies the patterns using DRR and it can serve as an example of such patterns for the future.

Reply:
@antiagainst I see, thanks!